compiler/src/dotty/tools/dotc/semanticdb/Tools.scala

```diff
@@ -12,7 +12,11 @@ object Tools:
   def mkURIstring(path: Path): String =
     // Calling `.toUri` on a relative path will convert it to absolute. Iteration through its parts instead preserves
     // the resulting URI as relative.
-    val uriParts = for part <- path.asScala yield new java.net.URI(null, null, part.toString, null)
+    // To prevent colon `:` from being treated as a scheme separator, prepend a slash `/` to trick the URI parser into
+    // treating the part as an absolute path, and then remove it afterwards.
+    val uriParts =
+      for part <- path.asScala
+      yield new java.net.URI(null, null, "/" + part.toString, null).toString().stripPrefix("/")
     uriParts.mkString("/")
```

Comment:
would it have the same result if you don't strip each individual part, and instead join the parts with `""` and then only strip the first prefix